www/utils/supabase/queries.ts

```diff
@@ -1,5 +1,41 @@
 import { cache } from 'react';
 import { SupabaseClient } from '@supabase/supabase-js';
+import { unstable_cache as next_unstable_cache } from 'next/cache';
+import { getFreeMessageCount } from './actions';
+
+export const unstable_cache = <Args extends any[], Output>(
+  callback: (...args: Args) => Promise<Output>,
+  key: string[],
+  options: { revalidate: number },
+) => {
+  return cache(
+    next_unstable_cache(
+      callback as unknown as (...args: any[]) => Promise<Output>,
+      key,
+      options
+    )
+  );
+};
+
```

Comment:
Should move this to its own utility file

Reply:
Gotcha.